File: shared/types/UserSummaryDTO.ts
// Interface til UserSummary som henter profilbilde, brukerid og fult navn
export interface UserSummaryDTO {
    id: number;
    fullName: string;
    profileImageUrl: string | null;

    // Ekstra logikk for samhandlig med andre brukere
    isFriend?: boolean;
    isBlocked?: boolean;
    hasBlockedMe?: boolean;
    lastUpdated?: number; // Unix timestamp
  }

  export enum GroupRequestStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
  Creator = 3
}

File: AFBack/DTOs/UserSummaryDto.cs
namespace AFBack.DTOs;
// denne bruker vi for å sende kun brukerens id, navn og profilbilde til å brukes i venneforespørsler, kommentarer, poster osv for å raskt se litt informasjon om brukeren.
// Kan legge til flere ting (skjønn, alder, osv) etterhvert
public class UserSummaryDto
{
    public string Id { get; set; } // Bruker-ID
    public string FullName { get; set; } = string.Empty; // Fornavn + etternavn (eller med mellomnavn)
    public string? ProfileImageUrl { get; set; } // Profilbilde (kan være null hvis ikke satt)
    
    // Logikk for samhandling mellom brukere
    public bool? IsFriend  { get; set; }
    public bool? IsBlocked { get; set; }
    
    public bool? HasBlockedMe { get; set; }
    public long? LastUpdated { get; set; }

}
